src/main/java/com/yourcodereview/jegors/task1/Main.java

```diff
@@ -38,21 +42,20 @@ public static void main(String[] args) {
         var startTime = Instant.now();
         LOGGER.log(INFO, "Execution start time: {0}", startTime);
 
-        process(path);
+        try (var ips = Files.lines(path)) {
+            System.out.println(countUnique(ips));
+        } catch (IOException e) {
+            LOGGER.log(ERROR, "Error during processing file: {0}", path);
+        }
 
         var executionTime = Duration.between(startTime, Instant.now());
         LOGGER.log(INFO, "Execution time: {0}", executionTime);
     }
 
-    private static void process(Path path) {
-        try (var lines = Files.lines(path)) {
-
-            var unique = lines.collect(countingUnique());
-
-            System.out.println(unique);
-
-        } catch (IOException e) {
-            LOGGER.log(ERROR, "Error during processing file: {0}", path);
-        }
+    private static long countUnique(Stream<String> ipAddresses) {
+        return ipAddresses
```

Comment:
@rabestro 
What about concurrency?  You doing it in single thread, so you read -> convert-> add to your storage. So, reading and conversion are independent tasks. You could create queue of read strings and handle it in separate thread and send them to your IP address container.